zcash_client_backend/src/data_api/mem_wallet.rs

```diff
@@ -236,7 +235,10 @@ impl WalletRead for MemoryWalletDb {
     }
 
     fn chain_height(&self) -> Result<Option<BlockHeight>, Self::Error> {
-        todo!()
+        match self.blocks.last_key_value() {
```

Comment:
Something to take into consideration here is how this is handled in `zcash_client_sqlite`: in that context, the chain tip may not yet have been scanned, so we get the current chain height as the maximum height known to the scan queue, rather than the blocks table.

Reply:
Blocks are added to the memory wallet database by `put_blocks` which currently receives a vector of `ScannedBlock`. Do we plan to change that ?

Reply:
Whatever is the case, ill open a separated PR for this.

Reply:
> Blocks are added to the memory wallet database by `put_blocks` which currently receives a vector of `ScannedBlock`. Do we plan to change that ?

No; the main distinction here is that the chain tip height is used in computing how to prioritize the scanning queue; we scan blocks out-of-order in many cases to improve time to spendability. The important related operation is `update_chain_tip`; basically, `chain_height` should return whatever is correct as of the last call to `update_chain_tip`, not the last call to `put_blocks`.

Reply:
In `update_chain_tip` where do you store the height that is passed by argument so you can use it in `chain_height`?